pulp_ansible/app/tasks/collections.py

```diff
@@ -608,13 +663,55 @@ async def _find_all_collections(self):
 
         await asyncio.gather(*tasks)
 
+    async def _can_be_optimized(self):
```

Comment:
I think this name is confusing. I think can be optimized is the user desire to optimize or not. This is really can we skip the sync or not. With that I'm suggesting:

```suggestion
    async def _should_we_sync(self):
```
Then invert the false and true statements to align with the logic called out by the name.